HIP/hip-1028.md

```diff
@@ -0,0 +1,157 @@
+---
+hip: 1028
+title: Metadata management via SmartContracts
+author: Mustafa Uzun <mustafa.uzun@limechain.tech>, Luke Lee <luke.lee@swirldslabs.com>
+working-group: Valentin Valkanov <valentin.valkanov@limechain.tech>, David Bakin <david.bakin@swirldsbals.com>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+last-call-date-time:
+created: 2024-08-15
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/pull/1028
+requires: 646, 657, 765
+updated: 2024-08-19
+requested-by: NA
+---
+
+## Abstract
+
+HIP-646/657/765 introduces a new metadata field for Fungible Tokens (FT), Non-Fungible Tokens (NFT) and METADATA key  
+for updating metadata. However, these features are not supported by smart contracts.  
+This proposal aims to enhance Hedera Smart Contract Service (HSCS) by exposing HIP-646/657/765 features.
+
+## Motivation
+
+A user may need to create, update and get info for Fungible Tokens (FT), Non-Fungible Tokens (NFT) and METADATA key through contract.  
```

Comment:
Specifically mention "token metadata and the metadata key"?

Reply:
suggestion: through contract -> from a smart contract.